github-auth-hook.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const { User, AuthenticationRequired } = require('unleash-server');
+
+const passport = require('passport');
+const GitHubStrategy = require('passport-github').Strategy;
+
+passport.use(
+    new GitHubStrategy(
+        {
+            clientID: process.env.GITHUB_CLIENT_ID,
+            clientSecret: process.env.GITHUB_CLIENT_SECRET,
+            callbackURL: process.env.GITHUB_CALLBACK_URL,
+        },
+
+        (accessToken, refreshToken, profile, done) => {
+            if (!profile.emails) {
+                // user can choose to not display any email, then use a default one as unleash required it
+                profile.emails.push(`${displayName}@unknown.com`);
+            }
+            done(
+                null,
+                new User({
+                    name: profile.displayName,
+                    email: profile.emails[0].value,
+                })
+            );
+        }
+    )
+);
+
+function enableGitHubOAuth(app) {
+
+    app.use(passport.initialize());
+    app.use(passport.session());
+
+    passport.serializeUser((user, done) => {
+        done(null, user);
+        });
+    passport.deserializeUser((user, done) => {     
+        done(null, user);
+        });
+ 
+    app.get('/api/admin/login', passport.authenticate('github'));
+    let context = process.env.TOGGLES_CONTEXT ? process.env.TOGGLES_CONTEXT : '';
+    app.get(
+        '/api/auth/callback',
+        passport.authenticate('github', {
+            failureRedirect: `${context}/api/admin/error-login`,
+        }),
+        (req, res) => {
+            // Successful authentication, redirect to your app.
+            res.redirect(`${context}/`);
```

Comment:
@aslakknutsen this redirect does not go through the proxy so it needs the context `/toggles/`